sale_order_lot_selection/tests/test_sale_order_lot_selection.py

```diff
@@ -0,0 +1,243 @@
+# © 2015 Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import openerp.tests.common as test_common
+from odoo.exceptions import Warning
+
+
+class TestSaleOrderLotSelection(test_common.SingleTransactionCase):
+
+    def setUp(self):
+        """
+        Set up a sale order a particular lot.
+
+        I confirm it, transfer the delivery order and check lot on picking
+
+        Set up a sale order with two lines with different products and lots.
+
+        I confirm it, transfer the delivery order and check lots on picking
+
+        """
+        super(TestSaleOrderLotSelection, self).setUp()
+        self.product_57 = self.env.ref('product.product_product_6')
+        self.product_46 = self.env.ref('product.product_product_17')
+        self.product_12 = self.env.ref('product.product_product_12')
+        self.supplier_location = self.env.ref(
+            'stock.stock_location_suppliers')
+        self.customer_location = self.env.ref(
+            'stock.stock_location_customers')
+        self.stock_location = self.env.ref('stock.stock_location_stock')
+        self.product_model = self.env['product.product']
+
+    def _stock_quantity(self, product, lot, location):
+        return product.with_context({
+            'lot_id': lot.id,
+            'location': location.id,
+        }).qty_available
+
+    def test_sale_order_lot_selection(self):
+        # INIT stock of products to 0
+        picking_out = self.env['stock.picking'].create({
+            'picking_type_id': self.env.ref('stock.picking_type_out').id,
+        })
+        self.env['stock.move'].create({
```

Comment:
`self.env['stock.move']` use multiple time, so you should use like a global variable